server/request_object.cc

```diff
@@ -288,6 +304,56 @@ void RequestObject::set_start_client_request_read_timeout() {
   return;
 }
 
+// Used in throttling S3 Get requests when there is memory issue
+int RequestObject::set_start_response_delay_timer(short int delay_in_milli_secs,
+                                                  void* action_obj) {
+  // Set the timer event when the size of outstanding response buffer
+  // to client crosses some threshold, causing memory shortage in S3 server.
+  s3_log(S3_LOG_DEBUG, request_id, "%s Entry\n", __func__);
+  if (!client_connected()) {
+    s3_log(S3_LOG_DEBUG, request_id, "Client is disconnected");
+    return -1;
+  }
+  struct timeval tv;
+  tv.tv_usec = delay_in_milli_secs * 1000;
+  tv.tv_sec = 0;
```

Comment:
need proper handling for cases when say user specifies 1500 milliseconds.  this has to set tv_usec = 500000 and tv_sec = 1.